lib/RWMutex.ts

```diff
@@ -227,21 +234,47 @@ export default class RWMutex {
    */
   async conditionalOverrideLockWriter(
     conditional: (oldWriter: string, newWriter: string) => Promise<boolean>,
-    upsert = false): Promise<boolean> { 
-    const mongoLock = await this._coll.findOne({ lockID: this._lockID })
-    if (!mongoLock) {
-      if (!upsert) {
-        return false;
+    upsert = true, timeout = 10000): Promise<boolean> { 
```

Comment:
It wasn't false for a specific reason before actually.  Just wanted to offer flexibility.  The reason we say upsert is true by default now has to do with the potential for a race condition that this PR addresses.  If we do a conditional check comparing the lock writers and then attempt to override the writer it's possible that the writer has changed between the check and our update.  In that case the query in our update call will not find a matching lock and we will by default attempt to create a new one.  Doing so will trigger a DuplicateKeyError because of our unique index on lockID.  If that happens we know to attempt the override again.  So, in short we use the upsert to trigger an error so that we continue to reattempt the override until the writer value we do the conditional check against matches the writer value of the lock we update.

Reply:
That makes sense! Thanks!